### PR TITLE
EdDSA gadget

### PR DESCRIPTION
**Pull request by yelhousni**
Reduce the number of constraints of a single EdDSA verification (e.g. Groth16/BN254 `-2049` constraints).

### CODE REVIEW

**Comment by yelhousni** [comment]
Modulo `Lookup2` for fixed inputs in PLONK #254

**Comment by ThomasPiellard** [comment]
So the error was not the lookup table but a wrong op in the sparse_r1cs constraint system:
`r.ModInverse(&r, q).Mul(&l, &r).Mod(&l, q) `instead of
`r.ModInverse(&r, q).Mul(&l, &r).Mod(&r, q)`

**Comment by gbotrel** [comment]
@yelhousni still a draft?

**Comment by yelhousni** [comment]
> @yelhousni still a draft?

PR is ready after Thomas' fix.

File: std/algebra/twistededwards/bandersnatch/point_test.go [hunk 1]
```diff
@@ -351,25 +361,39 @@ func TestNeg(t *testing.T) {
 
 }
 
-// benches
+// Bench
+func BenchmarkDouble(b *testing.B) {
```
Comment (gbotrel) [inline std/algebra/twistededwards/bandersnatch/point_test.go:365]: mmhh I see what we do here, not very go-idiomatic (ie the benchmark is going to be run many times).

One way is to add this to the go test command line: ` -benchtime=1x`

**Review by gbotrel — APPROVED** [review]

File: std/algebra/twistededwards/bandersnatch/point_test.go (hunk 1 above)
Comment (yelhousni) [reply std/algebra/twistededwards/bandersnatch/point_test.go:365]: Got rid of benchmarks. Can be ran with `api.Counter` when needed.

**Review by ThomasPiellard — APPROVED** [review]

**Pull request merged**